packages/mask/background/services/identity/profile/query.ts

```diff
@@ -28,3 +36,27 @@ export async function mobile_queryProfileRecordFromIndexedDB() {
     if (process.env.architecture !== 'app') throw new TypeError('This function is only available in app')
     return queryProfilesFromIndexedDB({})
 }
+
+export async function queryProfilesInformation(identifiers: ProfileIdentifier[]): Promise<ProfileInformation[]> {
+    const profiles = await queryProfilesDB({ identifiers })
+    return toProfileInformation(profiles)
+}
+/** @deprecated */
+export async function hasLocalKey(identifier: ProfileIdentifier) {
+    return hasLocalKeyOf(identifier)
+}
+
+export async function queryOwnedProfilesInformation(network?: string): Promise<ProfileInformation[]> {
+    let profiles: ProfileRecord[]
+    await createPersonaDBReadonlyAccess(async (t) => {
+        const personas = (await queryPersonasDB({ hasPrivateKey: true }, t)).sort((a, b) =>
+            a.updatedAt > b.updatedAt ? 1 : -1,
+        )
+        const ids = uniqBy(
+            personas.flatMap((x) => [...x.linkedProfiles.keys()]),
+            (x) => x.toText(),
+        )
+        profiles = await queryProfilesDB({ identifiers: ids, network }, t)
+    })
+    return toProfileInformation(profiles!.filter((x) => x.identifier.network === network))
```

Comment:
```suggestion
    let profiles: ProfileRecord[] = []
    await createPersonaDBReadonlyAccess(async (t) => {
        const personas = (await queryPersonasDB({ hasPrivateKey: true }, t)).sort((a, b) =>
            a.updatedAt > b.updatedAt ? 1 : -1,
        )
        const ids = uniqBy(
            personas.flatMap((x) => [...x.linkedProfiles.keys()]),
            (x) => x.toText(),
        )
        profiles = await queryProfilesDB({ identifiers: ids, network }, t)
    })
    return toProfileInformation(profiles.filter((x) => x.identifier.network === network))
```

runtime guarantee

Reply:
It's an impossible case. If it really happens, it means something wrong happens, it should throw instead of return an empty result.